classes/announcement/AnnouncementTypeDAO.php

```diff
@@ -98,108 +80,75 @@ public function updateLocaleFields($announcementType)
+        $announcementType->setId($id);
         $this->updateLocaleFields($announcementType);
-        return $announcementType->getId();
+        return $id;
     }
 
     /**
      * Update an existing announcement type.
-     *
-     * @param AnnouncementType $announcementType
-     *
-     * @return bool
      */
-    public function updateObject($announcementType)
+    public function updateObject(AnnouncementType $announcementType): bool
     {
-        $returner = $this->update(
-            'UPDATE	announcement_types
-                        SET	context_id = ?
-			WHERE	type_id = ?',
-            [
-                $announcementType->getContextId(),
-                (int) $announcementType->getId()
-            ]
-        );
+        $affected = DB::table('announcement_types')
+            ->where('type_id', '=', $announcementType->getId())
+            ->update(['context_id' => $announcementType->getContextId()]);
 
         $this->updateLocaleFields($announcementType);
-        return $returner;
+        return $affected > 0;
     }
 
     /**
-     * Delete an announcement type. Note that all announcements with this type are also
-     * deleted.
-     *
-     * @param AnnouncementType $announcementType
+     * Delete an announcement type.
+     * @param bool $deleteAnnouncements Defaults to false, when true, all announcements with this type are also deleted.
      */
-    public function deleteObject($announcementType)
+    public function deleteObject(AnnouncementType $announcementType, bool $deleteAnnouncements = false): void
     {
-        return $this->deleteById($announcementType->getId());
+        $this->deleteById($announcementType->getId(), $deleteAnnouncements);
     }
 
     /**
-     * Delete an announcement type by announcement type ID. Note that all announcements with
-     * this type ID are also deleted.
-     *
-     * @param int $typeId
+     * Delete an announcement type by announcement type ID. Note that all announcements with this type ID are also deleted.
+     * @param bool $deleteAnnouncements Defaults to false, when true, all announcements with this type are also deleted.
      */
-    public function deleteById($typeId)
+    public function deleteById(int $typeId, bool $deleteAnnouncements = false): void
     {
-        $this->update('DELETE FROM announcement_type_settings WHERE type_id = ?', [(int) $typeId]);
-        $this->update('DELETE FROM announcement_types WHERE type_id = ?', [(int) $typeId]);
+        if ($deleteAnnouncements) {
+            $collector = Repo::announcement()->getCollector()->filterByTypeIds([$typeId]);
+            Repo::announcement()->deleteMany($collector);
+        }
 
-        $collector = Repo::announcement()->getCollector()->filterByTypeIds([(int) $typeId]);
-        Repo::announcement()->deleteMany($collector);
+        DB::table('announcement_types')
+            ->where('type_id', '=', $typeId)
+            ->delete();
```

Comment:
Also the entries in the DB table announcement_type_settings would need to be deleted here, no?

Reply:
This should be handled by the database (there's an `on delete cascade`), and the only reason to delete manually the announcements above, is because there might be physical files on the disk, so an improvement would be to delete manually only the ones that have files.

Reply:
Ah, right, I forgot about on delete cascade here... :-)